### PR TITLE
Change nested formula causing memory allocations,add corresponding tests

### PR DESCRIPTION
**Pull request by hameye**
Fix: https://github.com/JuliaEarth/GeoStats.jl/issues/172

### CODE REVIEW

File: test/nested.jl [hunk 1]
```diff
@@ -4,14 +4,20 @@
   @test nugget(γ) ≈ 0.3
   @test sill(γ) ≈ 1.0
   @test range(γ) ≈ 50.0
+  @test (@elapsed sill(γ)) < 100e-6
```
Comment (juliohm) [inline test/nested.jl:7]: Shoudln't this be `1e-6`? I remember the new times were in ns (1e-9) right? We could make the test even tighter with `1e-7` for example.

File: test/nested.jl [hunk 2]
```diff
@@ -50,16 +58,22 @@
   @test γ(Point(10.,0.), Point(0.,0.)) ≈ sill(γ)
   @test isstationary(γ)
 
-  # test constructor explicitly
+  # test coe-6tructor explicitly
```
Comment (juliohm) [inline test/nested.jl:61]:
```suggestion
  # test constructor explicitly
```

**Review by juliohm — CHANGES_REQUESTED** [review]
Thanks! I think we just need to double check the elapsed time tests before merging. Could you please double check?

File: test/nested.jl (hunk 2 above)
Comment (hameye) [reply test/nested.jl:61]: Misuse of automatic replace, sorry for that ..

File: src/nested.jl [hunk 3]
```diff
@@ -27,9 +27,9 @@ end
 NestedVariogram(cs, γs) = NestedVariogram{typeof(cs),typeof(γs)}(cs, γs)
 
 # variogram interface
-(g::NestedVariogram)(h)                  = raw(sum(c*γ(h) for (c, γ) in zip(g.cs, g.γs)))
-(g::NestedVariogram)(x::Point, y::Point) = raw(sum(c*γ(x, y) for (c, γ) in zip(g.cs, g.γs)))
-sill(g::NestedVariogram)                 = raw(sum(c*sill(γ) for (c, γ) in zip(g.cs, g.γs)))
+(g::NestedVariogram)(h)                  = raw(sum(g.cs .* map(γ -> γ(h), g.γs)))
+(g::NestedVariogram)(x::Point, y::Point) = raw(sum(g.cs .* map(γ -> γ(x, y), g.γs)))
+sill(g::NestedVariogram)                 = raw(sum(g.cs .* map(sill, g.γs)))
 nugget(g::NestedVariogram)               = raw(sum(c*nugget(γ) for (c, γ) in zip(g.cs, g.γs)))
```
Comment (juliohm) [inline src/nested.jl:33]: Can you add a similar change to the `nugget`?

**Review by juliohm — CHANGES_REQUESTED** [review]
We missed the nugget method. We can introduce a similar change and add tests for it as well to make sure we never allocated unnecessarily.

**Review by juliohm — COMMENTED** [review]
Any reason not to add @elapsed tests for nugget as well?

File: test/nested.jl [hunk 4]
```diff
@@ -4,14 +4,26 @@
   @test nugget(γ) ≈ 0.3
   @test sill(γ) ≈ 1.0
   @test range(γ) ≈ 50.0
+  @test (@elapsed sill(γ)) < 1e-6
+  @test (@elapsed nugget(γ)) < 32
```
Comment (juliohm) [inline test/nested.jl:8]: You copied/pasted things incorrectly.

**Comment by juliohm** [comment]
Thanks! I will just wait for the tests on GitHub.

**Comment by juliohm** [comment]
It is interesting that the MacOS build failed with a time 2e-6. I will merge the improvement as is and we can work in the tolerances of the test suite later if necessary.

**Pull request merged**